cudax/test/device/device_smoke.cu

```diff
@@ -0,0 +1,238 @@
+    ::test_device_attribute<device::attrs::max_surface_2d_layered_height, ::cudaDevAttrMaxSurface2DLayeredHeight, int>();
+    ::test_device_attribute<device::attrs::max_surface_2d_layered_layers, ::cudaDevAttrMaxSurface2DLayeredLayers, int>();
+    ::test_device_attribute<device::attrs::max_surface_cubemap_width, ::cudaDevAttrMaxSurfaceCubemapWidth, int>();
+    ::test_device_attribute<device::attrs::max_surface_cubemap_layered_width,
+                            ::cudaDevAttrMaxSurfaceCubemapLayeredWidth,
+                            int>();
+    ::test_device_attribute<device::attrs::max_surface_cubemap_layered_layers,
+                            ::cudaDevAttrMaxSurfaceCubemapLayeredLayers,
+                            int>();
+    ::test_device_attribute<device::attrs::max_registers_per_block, ::cudaDevAttrMaxRegistersPerBlock, int>();
+    ::test_device_attribute<device::attrs::clock_rate, ::cudaDevAttrClockRate, int>();
+    ::test_device_attribute<device::attrs::texture_alignment, ::cudaDevAttrTextureAlignment, int>();
+    ::test_device_attribute<device::attrs::texture_pitch_alignment, ::cudaDevAttrTexturePitchAlignment, int>();
+    ::test_device_attribute<device::attrs::gpu_overlap, ::cudaDevAttrGpuOverlap, bool>();
+    ::test_device_attribute<device::attrs::multi_processor_count, ::cudaDevAttrMultiProcessorCount, int>();
+    ::test_device_attribute<device::attrs::kernel_exec_timeout, ::cudaDevAttrKernelExecTimeout, bool>();
+    ::test_device_attribute<device::attrs::integrated, ::cudaDevAttrIntegrated, bool>();
+    ::test_device_attribute<device::attrs::can_map_host_memory, ::cudaDevAttrCanMapHostMemory, bool>();
+    ::test_device_attribute<device::attrs::compute_mode, ::cudaDevAttrComputeMode, ::cudaComputeMode>();
+    ::test_device_attribute<device::attrs::concurrent_kernels, ::cudaDevAttrConcurrentKernels, bool>();
+    ::test_device_attribute<device::attrs::ecc_enabled, ::cudaDevAttrEccEnabled, bool>();
+    ::test_device_attribute<device::attrs::pci_bus_id, ::cudaDevAttrPciBusId, int>();
+    ::test_device_attribute<device::attrs::pci_device_id, ::cudaDevAttrPciDeviceId, int>();
+    ::test_device_attribute<device::attrs::tcc_driver, ::cudaDevAttrTccDriver, bool>();
+    ::test_device_attribute<device::attrs::memory_clock_rate, ::cudaDevAttrMemoryClockRate, int>();
+    ::test_device_attribute<device::attrs::global_memory_bus_width, ::cudaDevAttrGlobalMemoryBusWidth, int>();
+    ::test_device_attribute<device::attrs::l2_cache_size, ::cudaDevAttrL2CacheSize, int>();
+    ::test_device_attribute<device::attrs::max_threads_per_multi_processor,
+                            ::cudaDevAttrMaxThreadsPerMultiProcessor,
+                            int>();
+    ::test_device_attribute<device::attrs::unified_addressing, ::cudaDevAttrUnifiedAddressing, bool>();
+    ::test_device_attribute<device::attrs::compute_capability_major, ::cudaDevAttrComputeCapabilityMajor, int>();
```

Comment:
Note: Not necessarily relevant for this PR, but we will probably want to introduce types to represent different architectures. It would make sense to return them here.